### PR TITLE
Add measurement method and unit as parameter to GetStats API

### PR DESCRIPTION
**Pull request by shifucun**
This allows client to get a specific time series based on measurement method and unit. Only time series with exact exact match on given parameters are returned. If there are multiple of them, the regular ranking apply and returns the best ranked one.

### CODE REVIEW

File: proto/mixer.proto [hunk 1]
```diff
@@ -214,6 +214,12 @@ message GetStatsRequest {
   // The dcid of the StatisticalVariable.
   string stats_var = 2;
 
+  // The measurement method of the observation.
```
Comment (beets) [inline proto/mixer.proto:217]: please specify that these are optional here.

**Comment by beets** [comment]
lgtm, though how will this play out to the new proposal? will the other variants call this and keep these parameters optional?

File: proto/mixer.proto (hunk 1 above)
Comment (shifucun) [reply proto/mixer.proto]: Done

**Comment by shifucun** [comment]
> lgtm, though how will this play out to the new proposal? will the other variants call this and keep these parameters optional?

Yes, for "Simple" user case, python api can call this and apply filtering. Will still need the "Advanced" API which return multiple time series.

File: e2etest/get_stats_test.go [hunk 2]
```diff
@@ -48,27 +48,39 @@ func TestGetStats(t *testing.T) {
 	for _, c := range []struct {
 		statsVar     string
 		place        []string
+		mmethod      string
```
Comment (pradh) [inline e2etest/get_stats_test.go:51]: have a unit test?

Comment (shifucun) [reply e2etest/get_stats_test.go:51]: Done

Comment (pradh) [reply e2etest/get_stats_test.go:51]: Oh nevermind, later realized the unit isn't really implemented yet (blocked on cache).

**Comment by shifucun** [comment]
Thanks for review!

**Pull request merged**